lib/perso.ml

```diff
@@ -304,6 +304,24 @@ let limit_desc conf =
 
 let infinite = 10000
 
+let get_descendants_at_level base p lev2 =
+  match lev2 with
+  | 0 -> []
+  | n ->
+      let rec loop acc lev fam =
+        Array.fold_left
+          (fun acc ifam ->
+            let children = Array.to_list (get_children (foi base ifam)) in
+            List.fold_left
+              (fun acc ch ->
+                if lev < n then loop acc (lev + 1) (get_family (poi base ch))
+                else if not (List.mem ifam acc) then ifam :: acc
```

Comment:
this `List.mem` here makes the fold_left quadratic, you might want to use a Set here or do it differently

Reply:
I did not test it and I'm not sure if it gives better performance at all but you can try something like I did in this commit: https://github.com/geneweb/geneweb/commit/79e9a835d7e11936c729e4b41e0977b324e67c87